### DISCUSSION TITLE
«Zonemaps» concept

### DISCUSSION THREAD

**Discussion by aecorn** [duckdb/duckdb#7839] [Q&A]
After reading the duckdb documentation I found the concept of "zonemaps" there:
https://duckdb.org/docs/data/parquet/overview.html#partial-reading
This seems very compelling to me, but I've been unable to google "parquet zonemap" and find anything other relevant than the original link.

Is there some equivalent? Partitioning a parquetfile like you do with pyarrow? Where can I go to understand more about the concept and how to store files to actually contain these "zonemaps"?

**Comment by daz10000** [comment]
Now you find the original link and this ticket :)   A duckduckgo query gives me pictures of parquetry.  A short intro or link would be helpful.

**Comment by Tishj** [comment]
Perhaps this medium article helps?
It might not officially be called a zone map in the Parquet standard but in database architecture that is the term used for this

<https://medium.com/@diehardankush/predicate-pushdown-in-parquet-enhancing-efficiency-and-performance-5becb0b992de>

**Reply by gwerbin** [reply]
Sorry if this is a silly question, but it's not clear to me what metadata exactly is considered a "zonemap" by DuckDB for the purpose of using predicate pushdown.

If I have a dataset of partitioned Parquet files from an unknown source, is there something I can look for in the metadata to check if it has whatever DuckDB considers to be a zonemap? And/or can I check in a query plan somehow if pushdown into the file level is being applied?

**Reply by Tishj** [reply]
I think it's the stats
`stats_min`, `stats_max`
`stats_min_value`, `stats_max_value`

See <https://duckdb.org/docs/data/parquet/metadata>

**Comment by daz10000** [comment]
Thanks! - that's useful.  I guess it's an Oracle concept.  I'm still a little fuzzy on exactly what I need to do in practice building parquet files to get the most out of duckdb.  For example, focusing on construction of the index, or making the row groups align with the primary keys intuitively feels like it would help, but I don't see enough in docs to really know what to do.  It's impressively fast in practice so maybe I'm overthinking things.
